tgui/packages/tgui/interfaces/SpawnPanel/types.tsx

```diff
@@ -0,0 +1,46 @@
+export interface Data {
+  [panelName: string]: CreateObjectData[];
+}
+
+export enum SpawnPanelTabName {
+  createObject = 'Object',
+  createTurf = 'Turf',
+  createMob = 'Mob',
+}
+
+export type SpawnPanelTab = {
+  name: SpawnPanelTabName;
+  content: string;
+  icon: string;
+};
+
+export interface CreateObjectData {
+  Objects: {
+    [key: string]: {
+      icon: string;
+      icon_state: string;
+      name: string;
+      mapping: boolean;
+    };
+  };
+  Turfs: {
+    [key: string]: {
+      icon: string;
+      icon_state: string;
+      name: string;
+      mapping: boolean;
+    };
+  };
+  Mobs: {
+    [key: string]: {
+      icon: string;
+      icon_state: string;
+      name: string;
+      mapping: boolean;
+    };
```

Comment:
these shouldn't be nested, create a type for them, Objects: Record<string, Entity>, Turfs: Record<string, Entity>, etc